src/erlfmt.erl

```diff
@@ -58,7 +58,13 @@ init(State) ->
 
 %% API entry point
 -spec format_file(file:name_all() | stdin, config()) ->
-    {ok, [unicode:chardata()], [error_info()]} | {skip, string()} | {error, error_info()}.
+    {ok, #{
```

Comment:
We were thinking it might be better to extend config with a verbose prop
```erlang
-type config() :: [{pragma, pragma()} | {print_width, pos_integer()} | verbose].
```
then we can avoid checking the line length in cases where users didn't specify the verbose flag, to speed up the common case.

If there are warnings, we think it would also simplify things to then just append them to the single `warnings` list.

What do you think?

Reply:
Would it be too bad to keep the 3-element tuple here (i.e. `{ok, [unicode:chardata()], #{warnings => …}}`)?
When we use this function from `rebar3_format`, we just ignore the third element 🙄

Reply:
Yes we are also proposing to keep the 3 tuple, sorry if this was unclear.

Reply:
Done